src/e2e-test/features/pubsub/sink/GCSToPubSubAdditionalcases.feature

```diff
@@ -0,0 +1,236 @@
+
+  @CMEK @GCS_CSV_TEST
+  Scenario:Validate successful records transfer from GCS to PubSub using macro arguments with mandatory fields only
+    Given Open Datafusion Project to configure pipeline
+    When Source is GCS
+    When Sink is PubSub
+    Then Open GCS source properties
+    Then Enter GCS property reference name
+    Then Enter GCS property "projectId" as macro argument "gcsProjectId"
+    Then Enter GCS property "serviceAccountFilePath" as macro argument "gcsServiceAccount"
+    Then Enter GCS property "path" as macro argument "gcsSourcePath"
+    Then Enter GCS source property "skipHeader" as macro argument "gcsSkipHeader"
+    Then Enter GCS property "format" as macro argument "gcsFormat"
+    Then Enter GCS source property output schema "outputSchema" as macro argument "gcsOutputSchema"
+    Then Validate "GCS" plugin properties
+    Then Close the GCS properties
+    Then Open the PubSub sink properties
+    Then Enter PubSub property reference name
+    Then Enter PubSub property "projectId" as macro argument "PubSubProjectId"
+    Then Enter PubSub property "serviceAccountFilePath" as macro argument "PubSubServiceAccount"
+    Then Enter PubSub property "topic" as macro argument "PubSubSinkTopic"
+    Then Enter PubSub property "format" as macro argument "PubSubFormat"
+    Then Enter PubSub sink cmek property "encryptionKeyName" as macro argument "cmekPubSub" if cmek is enabled
+    Then Validate "GooglePublisher" plugin properties
+    Then Close the PubSub properties
+    Then Connect source as "GCS" and sink as "GooglePublisher" to establish connection
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "PubSubProjectId"
+    Then Enter runtime argument value "serviceAccountPath" for key "PubSubServiceAccount"
+    Then Enter runtime argument value "topic" for key "PubSubSinkTopic"
+    Then Enter runtime argument value "format" for key "PubSubFormat"
+    Then Enter runtime argument value "cmekPubSub" for PubSub Sink cmek property key "cmekPubSub" if PubSub Sink cmek is enabled
+    Then Enter runtime argument value "projectId" for key "gcsProjectId"
+    Then Enter runtime argument value "serviceAccountPath" for key "gcsServiceAccount"
+    Then Enter runtime argument value "gcsCsvFile" for GCS source property path key "gcsSourcePath"
+    Then Enter runtime argument value "gcsSkipHeaderTrue" for key "gcsSkipHeader"
+    Then Enter runtime argument value "csvFormat" for key "gcsFormat"
+    Then Enter runtime argument value "gcsCSVFileOutputSchema" for key "gcsOutputSchema"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+
+  @GCS_CSV_TEST
+  Scenario:Validate successful records transfer from GCS to PubSub using macro arguments with mandatory and non mandatory fields
```

Comment:
-> Validate successful records transfer from GCS to PubSub using advance fields with macro arguments